src/Build/BuildCheck/Infrastructure/ConfigurationProvider.cs

```diff
@@ -140,6 +147,74 @@ public static void CheckCustomConfigurationDataValidity(string projectFullPath,
         return configurations;
     }
 
+
+    /// <summary>
+    /// Generates a new dictionary that contains the key-value pairs from the original dictionary if the key starts with 'keyFilter'.
+    /// If updateKey is set to 'true', the keys of the new dictionary will not include keyFilter.
+    /// </summary>
+    /// <param name="keyFilter"></param>
+    /// <param name="originalConfiguration"></param>
+    /// <param name="updateKey"></param>
+    /// <returns></returns>
+    private Dictionary<string, string> FilterDictionaryByKeys(string keyFilter, Dictionary<string, string> originalConfiguration, bool updateKey = false)
+    {
+        var filteredConfig = new Dictionary<string, string>();
+
+        foreach (var kv in originalConfiguration)
+        {
+            if (kv.Key.StartsWith(keyFilter, StringComparison.OrdinalIgnoreCase))
+            {
+                var newKey = kv.Key;
+                if (updateKey)
+                {
+                    newKey = kv.Key.Substring(keyFilter.Length);
+                }
+                
+                filteredConfig[newKey] = kv.Value;
+            }
+        }
+
+        return filteredConfig;
+    }
+
+    /// <summary>
+    /// Fetches the .editorconfig data in form of Key-Value pair.
+    /// Resulted dictionary will contain only BuildCheck related rules.
+    /// </summary>
+    /// <param name="projectFullPath"></param>
+    /// <returns></returns>
+    /// <exception cref="BuildCheckConfigurationException"></exception>
+    private Dictionary<string, string> FetchEditorConfigRules(string projectFullPath)
+    {
+        // check if we have the data already
+        if (_editorConfigData.TryGetValue(projectFullPath, out var cachedConfig))
+        {
+            return cachedConfig;
+        }
+
+        Dictionary<string, string> config;
+        try
+        {
+            config = s_editorConfigParser.Parse(projectFullPath);
+        }
+        catch (Exception exception)
+        {
+            throw new BuildCheckConfigurationException($"Parsing editorConfig data failed", exception, BuildCheckConfigurationErrorScope.EditorConfigParser);
+        }
+
+        // clear the dictionary from the key-value pairs not BuildCheck related and
+        // store the data so there is no need to parse the .editorconfigs all over again
+        _editorConfigData[projectFullPath] = FilterDictionaryByKeys("build_check.",  config);
+
+        return _editorConfigData[projectFullPath];
```

Comment:
nit: No need to access the dictionary twice.

```suggestion
        Dictionary<string, string> result = FilterDictionaryByKeys("build_check.",  config);
        _editorConfigData[projectFullPath] = result;
        return result;
```